src/controllers/update/update.filing.date.controller.ts

```diff
@@ -1,22 +1,42 @@
 import { NextFunction, Request, Response } from "express";
 import { Session } from "@companieshouse/node-session-handler";
 
-import { logger } from "../../utils/logger";
+import { createAndLogErrorRequest, logger } from "../../utils/logger";
 import * as config from "../../config";
 import { isActiveFeature } from "../../utils/feature.flag";
-import { getApplicationData, setExtraData, mapFieldsToDataObject, mapDataObjectToFields } from "../../utils/application.data";
+import { getApplicationData, setExtraData, mapFieldsToDataObject, mapDataObjectToFields, addPrivateOverseasEntityDetailsToAppData } from "../../utils/application.data";
 import { postTransaction } from "../../service/transaction.service";
 import { createOverseasEntity, updateOverseasEntity } from "../../service/overseas.entities.service";
-import { OverseasEntityKey, Transactionkey, InputDateKeys } from '../../model/data.types.model';
+import { OverseasEntityKey, Transactionkey, InputDateKeys, EntityNumberKey } from '../../model/data.types.model';
 import { FilingDateKey, FilingDateKeys } from '../../model/date.model';
 import { ApplicationData } from "../../model/application.model";
+import { getPrivateOeDetails } from "../../service/private.overseas.entity.details";
 
-export const get = (req: Request, res: Response, next: NextFunction) => {
+const fetchAndSavePrivateOeData = async (req: Request, appData: ApplicationData): Promise<void> => {
+  const companyNumber = appData[EntityNumberKey];
+
+  if (!companyNumber) {
+    throw createAndLogErrorRequest(req, "Unable to request private OE data. Missing entity number in application data");
+  }
+
+  const privateOeDetails = await getPrivateOeDetails(req, companyNumber);
+
+  if (!privateOeDetails) {
+    throw createAndLogErrorRequest(req, "Request failed retrieving private OE data. Entity number not found");
+  }
+
+  addPrivateOverseasEntityDetailsToAppData(req.session, appData, privateOeDetails);
+};
+
+export const get = async(req: Request, res: Response, next: NextFunction) => {
   try {
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
     const appData = getApplicationData(req.session);
     const filingDate = appData.update?.[FilingDateKey] ? mapDataObjectToFields(appData.update[FilingDateKey], FilingDateKeys, InputDateKeys) : {};
 
+    await fetchAndSavePrivateOeData(req, appData);
```

Comment:
If the user edits the email address but goes back to this page the email will be fetched again. Overwriting the user edit. Email addr needs to be store in OE entity part of model and there should be a check to see if it is already fetched (not null or empty in OE entity) and only fetch if not.

Reply:
Good catch :thumbsup: We now only fetch if it hasn't been already.